honeybadger/tests/contrib/test_django.py

```diff
@@ -64,44 +92,79 @@ def test_generate_payload_with_session(self):
+        self.url = url(r'test', plain_view, name='test_view')
 
     def tearDown(self):
         clear_request()
 
-    def test_process_request(self):
-        request = self.rf.get('/test')
-        request.resolver_match = self.url.resolve('test')
-        self.middleware.process_request(request)
-        self.assertEqual(request, current_request())
+    def get_response(self, request):
+        return Mock()
 
     @patch('honeybadger.contrib.django.honeybadger')
     def test_process_exception(self, mock_hb):
         request = self.rf.get('test')
         request.resolver_match = self.url.resolve('test')
-        self.middleware.process_request(request)
         exc = ValueError('test exception')
-        self.middleware.process_exception(request, exc)
+
+        middleware = DjangoHoneybadgerMiddleware(self.get_response)
+        middleware.process_exception(request, exc)
 
         mock_hb.notify.assert_called_with(exc)
         self.assertIsNone(current_request(), msg='Current request should be cleared after exception handling')
 
-    def test_process_response(self):
+    def test___call__(self):
         request = self.rf.get('test')
         request.resolver_match = self.url.resolve('test')
-        response = Mock()
-        honeybadger.set_context(foo='bar')
-        self.middleware.process_request(request)
 
-        self.middleware.process_response(request, response)
+        middleware = DjangoHoneybadgerMiddleware(self.get_response)
+        response = middleware(request)
+
         self.assertDictEqual({}, honeybadger._get_context(), msg='Context should be cleared after response handling')
-        self.assertIsNone(current_request(), msg='Current request should be cleared after response handling')
\ No newline at end of file
+        self.assertIsNone(current_request(), msg='Current request should be cleared after response handling')
+
+
+@override_settings(
+    ROOT_URLCONF='honeybadger.tests.contrib.django_test_app.urls',
+    MIDDLEWARE=['honeybadger.contrib.django.DjangoHoneybadgerMiddleware']
+)
+class DjangoMiddlewareIntegrationTestCase(SimpleTestCase):
+    def setUp(self):
+        self.client = Client()
+
+    @unittest.skipUnless(versions_match(), "Current Python version unsupported by current version of Django")
+    def test_context_cleared_after_response(self):
+        self.assertIsNone(current_request(), msg='Current request should be empty prior to request')
+        response = self.client.get('/plain_view')
+        self.assertIsNone(current_request(), msg='Current request should be cleared after request processed')
+
+    @unittest.skipUnless(versions_match(), "Current Python version unsupported by current version of Django")
+    @override_settings(
+        HONEYBADGER={
+            'API_KEY': 'abc123'
+        }
+    )
+    def test_exceptions_handled_by_middleware(self):
+        def assert_payload(req):
+            error_payload = json.loads(req.data)
```

Comment:
```suggestion
            error_payload = json.loads(str(req.data, "utf-8"))
```

This fixes the test failures on Python 3.5 for me. Not sure if this is the idiomatic way to do this.